packages/generate/src/queries.ts

```diff
@@ -158,7 +165,33 @@ export function stringifyImports(imports: ImportMap) {
     .join("\n");
 }
 
-async function getMatches(root: string, schemaDir: string) {
+async function getMatches(
+  root: string,
+  schemaDir: string,
+  patterns?: string[],
+) {
+  if (patterns && patterns.length > 0) {
+    // Use globby to match files with the provided patterns
+    const { globby } = await import("globby");
```

Comment:
Since this would not be conditional anymore, we can just move this into a normal import.